dev/benchmarks/test_apps/stocks/lib/stock_state.dart

```diff
@@ -0,0 +1,187 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'stock_data.dart';
+
+@immutable
+abstract class StockRoutePath { const StockRoutePath(); }
+class StockHomePath extends StockRoutePath { const StockHomePath(); }
+class StockSettingsPath extends StockRoutePath { const StockSettingsPath(); }
+class StockSymbolPath extends StockRoutePath {
+  const StockSymbolPath(this.symbol);
+  final String symbol;
+}
+
+enum StockMode { optimistic, pessimistic }
+enum BackupMode { enabled, disabled }
+
+class StockConfiguration {
```

Comment:
we could leave this in stock_types.dart to make the patch simpler; maybe I'm missing a good reason to move it here though?

Reply:
Isn't this part of the appstate? wouldn't it be good to group all the app state together in one file?

Reply:
I separated out the router state to be in the routing/router_state.dart

Reply:
+1 to keeping this in the old file for now.